testdata/union/union.go

```diff
@@ -5,7 +5,6 @@ type UnionConstraint[T string | int64] struct {
 }
 
 // Repeated constraints are redundant
-// TODO: Write a mutation to remove redundant constraints
 type Repeated[T string | string | int64 | uint64] struct {
 	Value T
 }
```

Comment:
Let's add a non-literal type as well, as the simplify change only dedupes literals. An eventual change might be to dedupe these as well :shrug: 

```suggestion
type Foo struct{}

type Repeated[T string | string | int64 | uint64 | Foo | Foo] struct {
	Value T
}
```

Reply:
I tried that but Go failed to type-check because it's invalid code. Can't define the same type twice as generic constraint

Reply:
TIL :+1: